backend/src/modules/azure/schedules/synchronize-ad-users.cron.azure.use-case.ts

```diff
@@ -91,28 +116,51 @@ export class SynchronizeADUsersCronUseCase implements SynchronizeADUsersCronUseC
 			}
 		}
 	}
-	private async addUsersToApp(usersADFiltered: Array<AzureUserDTO>, usersApp: Array<User>) {
+	private async addUsersToApp(
+		usersADFiltered: Array<AzureUserDTO>,
+		usersApp: Array<User>,
+		team: Team
+	) {
 		const notIntersectedUsers = usersADFiltered.filter(
 			(userAd) =>
 				usersApp.findIndex(
 					(userApp) => userApp.email === (userAd.mail ?? userAd.userPrincipalName)
 				) === -1
 		);
 
-		for (const user of notIntersectedUsers) {
-			try {
+		try {
+			const usersToCreate: Array<CreateUserAzureDto> = notIntersectedUsers.map((user) => {
 				const splittedName = user.displayName.split(' ');
-				await this.createUserService.create({
-					email: user.mail,
+
+				return {
+					email: user.mail ?? user.userPrincipalName,
 					firstName: splittedName[0],
-					lastName: splittedName.at(-1),
+					lastName: splittedName[-1],
 					providerAccountCreatedAt: user.createdDateTime
-				});
-			} catch (err) {
-				this.logger.error(
-					`An error as occurred while creating user with email '${user.mail}' through the syncronize AD Users Cron. Message: ${err.message}`
-				);
+				};
+			});
+
+			const createdUsers = await this.createUserService.createMany(usersToCreate);
+
+			if (typeof team !== 'undefined') {
```

Comment:
```suggestion
			if (team) {
```

Do we really need to check the type of the team?

Reply:
Force of habit, in this case bot are equally fine to use.